### PR TITLE
Alerting: Swap order between Annotations and Labels step in the alert rule form.

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR change the order in steps in the alert rule form and updates some texts. (Same change is done in the Modfiy export form)
This PR also fix data source and panel label not being correct.

**_Before the change we had:_**
4. Add annotations 
5. Labels and notifications

**_After the change:_**
4. Configure labels and notifications
5. Add annotations

**Why do we need this feature?**

We need to provide the best user experience, and with this change we think users can go through the steps in a more logical way.

**Who is this feature for?**

All users.

**Special notes for your reviewer:**

Screenshot 2024-01-23 at 13 37 21

**_Recording rules are not affected:_**

Screenshot 2024-01-23 at 13 00 09

**_Before the data source and panel label fix:_**

Screenshot 2024-01-23 at 15 34 43

**_After the data source and panel label fix:_**

Screenshot 2024-01-23 at 15 34 09

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by soniaAguilarPeiron** [comment]
I think we should update docs , right? @brendamuir

**Review by brendamuir — APPROVED** [review]

**Review by dhalachliyski — APPROVED** [review]

File: public/app/features/alerting/unified/components/rule-editor/AnnotationHeaderField.tsx [hunk 1]
```diff
@@ -38,8 +38,10 @@ const AnnotationHeaderField = ({
               switch (annotationField.key) {
                 case Annotation.dashboardUID:
                   label = 'Dashboard and panel';
+                  break;
                 case Annotation.panelID:
                   label = '';
+                  break;
                 default:
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-editor/AnnotationHeaderField.tsx:45]: this was a 🐛

Comment (konrad147) [reply public/app/features/alerting/unified/components/rule-editor/AnnotationHeaderField.tsx:45]: nice catch!

Comment (konrad147) [reply public/app/features/alerting/unified/components/rule-editor/AnnotationHeaderField.tsx:45]: I wonder why `dashboardUID` has `Dashboard and panel` label while `panelID` is empty?

**Review by konrad147 — APPROVED** [review]
LGTM! one question in the review

**Pull request merged**